### PR TITLE
Improve goal feature

### LINKED ISSUE #124: Improve Goal feature
- Allow removing a book's `goal`
- Display `In Progress`, `Completed` and `Overdue` goal in Detailed view (KIV: Colored Goal display)
- Fix bug: `Goal` can be set with page > `totalPages`
- Fix bug: `edit` causes removal of goal

### PR DESCRIPTION
**Pull request by mgiang2015**
closes #124 

- Fix bug: Reader cannot set `Goal` to be higher than `totalPages` now.
- New features:
1. Ui: Goal completed, in progress and overdue.
1. `goaldel INDEX`: remove goal

### CODE REVIEW

File: src/test/java/seedu/bookmark/logic/commands/GoalDelCommandTest.java [hunk 1]
```diff
@@ -0,0 +1,73 @@
...
+
+import seedu.bookmark.commons.core.Messages;
+import seedu.bookmark.commons.core.index.Index;
+import seedu.bookmark.model.Model;
+import seedu.bookmark.model.ModelManager;
+import seedu.bookmark.model.UserPrefs;
+import seedu.bookmark.model.book.Book;
+import seedu.bookmark.model.book.Goal;
+
+public class GoalDelCommandTest {
```
Comment (TanLeYang) [inline src/test/java/seedu/bookmark/logic/commands/GoalDelCommandTest.java:21]: Maybe one more test for if the book has no goals then you try to delete the goal?

**Review by TanLeYang — COMMENTED** [review]
LGTM other than perhaps including 1 more test for GoalDelCommandTest!

File: src/test/java/seedu/bookmark/logic/commands/GoalDelCommandTest.java (hunk 1 above)
Comment (mgiang2015) [reply src/test/java/seedu/bookmark/logic/commands/GoalDelCommandTest.java:21]: Thanks for the suggestion! Will add that in

**Review by TanLeYang — APPROVED** [review]
LGTM!

**Pull request merged**